Core/HLE/sceKernelInterrupt.cpp

```diff
@@ -826,6 +826,63 @@ static u32 sysclib_strncpy(u32 dest, u32 src, u32 size) {
 	return hleLogSuccessX(SCEKERNEL, dest);
 }
 
+static u32 sysclib_strtol(u32 t, u32 endStringtemp, int base) {
+	ERROR_LOG(SCEKERNEL, "Unimp sysclib_strtol(%08x, %08x, %i)", t ,endStringtemp, base);
+	// base == 0 seems to be handled as base == 10.
+	if (base == 0){
+		base = 10;
+	}
+	std::string str = Memory::GetCharPointer(t);
+	std::string endString = Memory::GetCharPointer(endStringtemp);
+
+	// Skip any leading "0x" in case of base 16
+	if (base == 16 && (str.rfind("0x", 0) == 0 || str.rfind("0X", 0) == 0)) {
+		str = str.substr(2);
+	}	
+	return 0;
+}
+
+static u32 sysclib_strchr(u32 src, int c) {
+	const std::string str = Memory::GetCharPointer(src);
+	ERROR_LOG(SCEKERNEL, "Untest sysclib_strchr(%c, %i)", str,c);
+	int index = str.find(str,c);
+	if (index < 0) {
+		return 0;
+	}
+	return src + index;
+}
+
+static u32 sysclib_strrchr(u32 src, int c) {
+	const std::string str = Memory::GetCharPointer(src);
+	ERROR_LOG(SCEKERNEL, "Untest sysclib_strchr(%c, %i)", str, c);
+	int index = str.rfind(str, c);
+	if (index < 0) {
+		return 0;
+	}
+	return src + index;
+}
+
+const u8 look_ctype_table[128] = { 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x08, 0x08, 0x08, 0x08, 0x08, 0x20, 0x20,
+		0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20, 0x20,
+		0x18, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10,
+		0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x04, 0x10, 0x10, 0x10, 0x10, 0x10, 0x10,
+		0x10, 0x41, 0x41, 0x41, 0x41, 0x41, 0x41, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01,
+		0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x10, 0x10, 0x10, 0x10, 0x10,
+		0x10, 0x42, 0x42, 0x42, 0x42, 0x42, 0x42, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02,
+		0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x02, 0x10, 0x10, 0x10, 0x10, 0x20
+};
+
+static u32 sysclib_toupper(u32 c) {
+	ERROR_LOG(SCEKERNEL, "Untest sysclib_toupper(%i)",c);
+	int ctype = look_ctype_table[c & 0xFF];
```

Comment:
The table is only `[128]` so this can go out of bounds.  We could improve this to do something if `(c & 0xFF) >= 128`, or we could just use native `toupper()`.  Both likely don't support Unicode.

-[Unknown]